bodo/io/arrow.cpp

```diff
@@ -82,16 +82,12 @@ ArrowReader* snowflake_reader_init_py_entry(
 
 // --------- functions defined in parquet_write.cpp ---------
 int64_t pq_write_py_entry(const char* _path_name, table_info* table,
-                          array_info* col_names_arr, array_info* index,
-                          bool write_index, const char* metadata,
+                          array_info* col_names_arr, const char* metadata,
                           const char* compression, bool is_parallel,
-                          bool write_rangeindex_to_metadata, const int ri_start,
-                          const int ri_stop, const int ri_step,
-                          const char* idx_name, const char* bucket_region,
-                          int64_t row_group_size, const char* prefix,
-                          bool convert_timedelta_to_int64, const char* tz,
-                          bool downcast_time_ns_to_us, bool create_dir,
-                          bool force_hdfs);
+                          const char* bucket_region, int64_t row_group_size,
+                          const char* prefix, bool convert_timedelta_to_int64,
+                          const char* tz, bool downcast_time_ns_to_us,
+                          bool create_dir, bool force_hdfs);
```

Comment:
Happy to trim a lot of junk about index handling from this C++ function, especially since its reused for Iceberg and Snowflake that both don't care abound indexes at all

Reply:
Great to see!